hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java

```diff
@@ -70,120 +61,83 @@
-		if (!ledger.exists(target) || ledger.isSmartContract(target)) {
-			return INVALID_ACCOUNT_ID;
+		/* --- Translate from gRPC types --- */
+		final var op = txnCtx.accessor().getTxn().getCryptoUpdateAccount();
+
+		/* --- Load the model objects --- */
+		final var accountId = Id.fromGrpcAccount(op.getAccountIDToUpdate());
+		final var account = accountStore.loadPossiblyDetachedAccount(accountId);
+
+		final var newKey = op.hasKey() ? op.getKey() : null;
+		final var newMemo = op.hasMemo() ? op.getMemo().getValue() : null;
+		final var newAutoRenewPeriod = op.hasAutoRenewPeriod() ? op.getAutoRenewPeriod() : null;
+		final var newExpirationTime = op.hasExpirationTime() ? op.getExpirationTime() : null;
+		final var newProxyAccountID = op.hasProxyAccountID() ? op.getProxyAccountID() : null;
+		final var newReceiverSigRequiredWrapper = op.hasReceiverSigRequiredWrapper() ? op.getReceiverSigRequiredWrapper() : null;
+		final var newMaxAutomaticTokenAssociations = op.hasMaxAutomaticTokenAssociations() ? op.getMaxAutomaticTokenAssociations() : null;
+
+		if (account.isDetached()) {
+			validateTrue(isExpiryOnly(op), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
 		}
-
-		final var changes = customizer.getChanges();
-		final var keyChanges = customizer.getChanges().keySet();
-
-		if (ledger.isDetached(target)) {
-			if (!keyChanges.equals(EXPIRY_ONLY)) {
-				return ACCOUNT_EXPIRED_AND_PENDING_REMOVAL;
-			}
-		}
-
-		if (keyChanges.contains(AccountProperty.EXPIRY)) {
-			final long newExpiry = (long)changes.get(AccountProperty.EXPIRY);
-			if (newExpiry < ledger.expiry(target)) {
-				return EXPIRATION_REDUCTION_NOT_ALLOWED;
-			}
+		if (newExpirationTime != null) {
+			validateTrue(validator.isValidExpiry(newExpirationTime), INVALID_EXPIRATION_TIME);
+			validateTrue(newExpirationTime.getSeconds() > account.getExpiry(), EXPIRATION_REDUCTION_NOT_ALLOWED);
 		}
-
-		if (keyChanges.contains(AccountProperty.MAX_AUTOMATIC_ASSOCIATIONS)) {
-			final long newMax = (int) changes.get(AccountProperty.MAX_AUTOMATIC_ASSOCIATIONS);
-			if (newMax < ledger.alreadyUsedAutomaticAssociations(target)) {
-				return EXISTING_AUTOMATIC_ASSOCIATIONS_EXCEED_GIVEN_LIMIT;
-			}
-			if (newMax > dynamicProperties.maxTokensPerAccount()) {
-				return REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT;
-			}
+		if (newMaxAutomaticTokenAssociations != null) {
+			validateTrue(newMaxAutomaticTokenAssociations.getValue() >= account.getAlreadyUsedAutomaticAssociations(), EXISTING_AUTOMATIC_ASSOCIATIONS_EXCEED_GIVEN_LIMIT);
+			validateTrue(newMaxAutomaticTokenAssociations.getValue() <= dynamicProperties.maxTokensPerAccount(), REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT);
 		}
 
-		return OK;
+		/* --- Do the business logic --- */
+		account.updateFromGrpc(
+				newKey,
+				newMemo,
+				newAutoRenewPeriod,
+				newExpirationTime,
+				newProxyAccountID,
+				newReceiverSigRequiredWrapper,
+				newMaxAutomaticTokenAssociations);
```

Comment:
If we pass these parameters as Optional.of(X) we might be able to skip a the ternary operators above.